src/components/PasswordInput.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+    <label>
+        <div class="input__wrapper">
+            <span
+                v-if="label"
+                class="base-input__label">
+                {{ label }}
+            </span>
+            
+            <input 
+                :value="value"
+                @input="$emit('input', $event.target.value)"
+                :placeholder="placeholder"
+                :type="changeType"
+                class="base-input__input"> 
+                
+                <button title="Show password" class="button__password" @click="ShowHidePassword">
+                    <img src="@/assets/eyeCrossed.svg" alt="showPassword" :class="visible">
+                    <img src="@/assets/eye.svg" alt="hidePassword" class="hide" :class="unvisible">
+                </button>
+            
+        </div>
+    </label>
+</template>
+
+<script>
+    export default {
+        props: {
+            value: {
+                type: String,
+                default: ''
+            },
+            label: {
+                type: String,
+                default: ''
+            },
+            placeholder: {
+                type: String,
+                default: ''
+            },
+            type: {
+                type: String,
+                default: 'password'
+            }
+        },
+        data() {
+            return {
+                showPassword: false,
+                visibleEye: true,
+                unvisibleEye: true
+            }
+        },
+        methods: {
+            ShowHidePassword(event) {
+                event.preventDefault();
+                this.showPassword = !this.showPassword;
+                this.visibleEye = !this.visibleEye;
+                this.unvisibleEye = !this.unvisibleEye;
+            }
+        },
+        computed: {
+            changeType() {
```

Comment:
name `changeType` suggests it's a method to change a type, but it's not. What do you say about changing it to something that describes what it is e.g. `inputType`?